_episodes_rmd/03-r-data-analysis.Rmd

```diff
@@ -33,258 +32,315 @@ knitr_fig_path("03-")
 > >
 > > ```{r FilterChallenge1}
-> > gapminder_data %>%
-> > summarize(first_year=min(year))
+> > smoking %>%
+> > summarize(recent_year=max(year))
 > > ```
 > >
-> > We see here that the first year in the dataset is 1952. Filter to only 1952, and determin the average GDP per capita.
+> > Filter to only the maximum year, and determine the average percent of people with lung cancer.
 > >
 > > ```{r FilterChallenge2}
-> > gapminder_data %>%
-> > filter(year == 1952) %>%
-> > summarize(average_gdp=mean(gdpPercap))
+> > smoking %>%
+> > filter(year == 2019) %>%
+> > summarize(mean_cancer_pct=mean(lung_cancer_pct))
 > > ```
 > > {: .source}
-> > By combining `filter()` and `summarize()` we were able to calculate the mean GDP per capita in the year 1952.
+> > By combining `filter()` and `summarize()` we were able to calculate the mean mercent of people with lung cancer in 2019
+> > 
+> > Bonus:
+> > Instead of finding the minimum year first and then filtering, we can find the minimum year right in the filter function: 
+> > ```{r no_hard_code}
+> > smoking %>% 
+> > filter(year == min(year)) %>% 
+> > summarize(mean_cancer_pct=mean(lung_cancer_pct))
+> > ```
 > {: .solution}
 {: .challenge}
 
-Notice how the pipe operator (`%>%`) allows us to combine these two simple steps into a more complicated data extraction?. We took the data, filtered out the rows, then took the mean value. The argument we pass to `filter()` needs to be some expression that will return TRUE or FALSE. We can use comparisons like `>` (greater than) and `<` (less than) for example. Here we tested for equality using a double equals sign `==`. You use `==` (double equals) when testing if two values are equal, and you use `=` (single equals) when naming arguments that you are passing to functions. Try changing it to use `filter(year = 2007)` and see what happens.
-
 ## Grouping rows using `group_by()` {#grouping-rows-using-group_by}
-
 [*Back to top*](#contents)
 
-We see that the life expectancy in 2007 is much larger than the value we got using all of the rows. It seems life expectancy is increasing which is good news. But now we might be interested in calculating the average for each year. Rather that doing a bunch of different `filter()` statements, we can instead use the `group_by()` function. The function allows us to tell the code to treat the rows in logical groups, so rather than summarizing over all the rows, we will get one summary value for each group. Here's what that will look like:
+We see that the mean percent of smokers is larger in 1990 than the value we got using all of the rows. But now we might be interested in calculating the average for each year. Rather that doing a bunch of different `filter()` statements, we can instead use the `group_by()` function. The function allows us to tell the code to treat the rows in logical groups, so rather than summarizing over all the rows, we will get one summary value for each group. Here's what that will look like:
```

Comment:
```suggestion
We previously saw that the mean percent of smokers is larger in 1990 than the value we got using all of the rows. But now we might be interested in calculating the average for each year. Rather that doing a bunch of different `filter()` statements, we can instead use the `group_by()` function. The function allows us to treat the rows in logical groups, so rather than summarizing over all the rows, we will get one summary value for each group. Here's what that will look like:
```